src/Microsoft.Diagnostics.Monitoring.WebApi/CollectionRuleStateHolder.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    internal class CollectionRuleStateHolder
+    {
+        public CollectionRuleState CurrentState { get; private set; } = CollectionRuleState.Running;
+        public string CurrentStateReason { get; private set; } = Strings.Message_CollectionRuleStateReason_Running;
+
+        private readonly object _lock = new object();
+
+        public CollectionRuleStateHolder(CollectionRuleStateHolder other)
+        {
+            lock (_lock)
```

Comment:
This is in the ctor so it doesn't make sense to lock itself. What you really want is to lock the other's _lock field since you are copying from it: `lock (other._lock)`